artichoke-backend/src/extn/core/time/mruby.rs

```diff
@@ -112,10 +112,11 @@ unsafe extern "C" fn time_self_now(mrb: *mut sys::mrb_state, _slf: sys::mrb_valu
 }
 
 unsafe extern "C" fn time_self_at(mrb: *mut sys::mrb_state, _slf: sys::mrb_value) -> sys::mrb_value {
-    let args = mrb_get_args!(mrb, *args);
+    let (seconds, microseconds) = mrb_get_args!(mrb, required = 1, optional = 1);
     unwrap_interpreter!(mrb, to => guard);
-    let args = args.iter().copied().map(Value::from);
-    let result = trampoline::at(&mut guard, args);
+    let seconds = Value::from(seconds);
+    let microseconds = Value::from(microseconds);
```

Comment:
since `microseconds` is an optional argument, the `mrb_get_args!` macro returns an `Option<sys::mrb_value>`.

We can preserve this type information and pass an `Option<Value>` into the trampoline, which avoids the `is_nil` check by doing this:

```suggestion
    let microseconds = microseconds.map(Value::from);
```